research-software-directory/RSD-as-a-service
# 252 edit project team

# Edit project team

Closes #252
Closes #164

Changes proposed in this pull request:
* Project maintainer add/edit/delete team members using find/add options
* We search in RSD and ORCID database
* If person extracted from ORCID has more than one affiliation we show them in drop-down. User still can type in completely different affiliation. Same solution is implemented the edit contributors modal.

How to test:
* `docker-compose down --volumes`: remove old
* `docker-compose build`: to build solution
* `docker-compose up`: to start
* data-migration is optional
* login using any of test users, eg. professor1
* create new project
* add team members using search on the right side of the screen, the section should look like in the example below
* when using information from ORCID, which might provide multiple affiliation values, first affiliation from the array is used and the other values can be selected from dropdown. To make clear that multiple options are present, the dropdown will open on focus. The help text is updated to state "Select or type in...". Beside selecting the dropdown options the user can type in any text value.

![image](https://user-images.githubusercontent.com/9204081/167309208-e6bc7b24-e201-453a-a8f6-ffad182a930a.png)

## Affiliation from ORCID example

![image](https://user-images.githubusercontent.com/9204081/167579166-4b7aec57-9ff0-47f7-af10-a50307753af2.png)


PR Checklist:
